### PR TITLE
Refactor all require decorators to use skipUnless when possible

### PR DESCRIPTION
**Pull request by muellerzr**
# What does this PR do?

I was refactoring the Accelerate tests today and I noticed we use `if .. else..` as a conditional for skipping tests based on imports. Unittest has `skipUnless` and `skipIf`, letting us simplify those decorators to be one line. 

E.g.:

```python
    if not _run_slow_tests:
        return unittest.skip("test is slow")(test_case)
    else:
        return test_case
```
Can be:
```python
    return unittest.skipUnless(_run_slow_tests, "test is slow")(test_case)
```

(Adding you as a reviewer for this Sylvain, unsure who else should be added)

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for the simplification!
Let's also wait for @LysandreJik approval.

**Review by LysandreJik — APPROVED** [review]
LGTM, pinging @ydshieh for knowledge.

**Review by ydshieh — APPROVED** [review]

**Pull request merged**
